analyses/cell-type-ewings/exploratory_analysis/06-tumor-cell-state-assignment.Rmd

```diff
@@ -0,0 +1,606 @@
+      library_id == "SCPCL000824" & cluster == 9 ~ "macrophage",
+      # cluster 1 has high MSC marker gene expression and a mix of MSC like cell types
+      library_id == "SCPCL000824" & cluster == 1 ~ "mesenchymal-like cell",
+      # if they didn't have high expression of any markers, we label them as unknown
+      .default = "unknown"
+    )
+  )
+```
+
+```{r}
+library_ids |>
+  purrr::map(\(id){
+    cluster_classification_df |>
+      dplyr::filter(library_id == id) |>
+      ggplot(aes(x = UMAP1, y = UMAP2, color = cluster_classification)) +
+      geom_point(alpha = 0.5, size = 0.1) +
+      theme(
+        aspect.ratio = 1
+      ) +
+      labs(title = id) +
+      scale_color_brewer(palette = "Dark2") +
+      guides(color = guide_legend(override.aes = list(alpha = 1, size = 1.5)))
+  })
+```
+
+Looking at this I think the cells that are classified as tumor cells using the cluster assignments mostly line up with the `SingleR` annotations. 
+So we can label any cells from clusters that have high tumor marker gene expression as tumor cells. 
+
+### Look at marker gene expression for tumor cell states in all cells
+
+Below we look at the expression of genes for the tumor cell states across all cells. 
+This will show us how the expression of these gene sets varies across cells in each library and help ensure we have the correct population of tumor cells.
+
+The value for each of the marker gene sets corresponds to the sum of the expression of all genes in the given gene list. 
+(Note that I also looked at mean and it had similar output)
+
+The gene sets that we use are: 
+
+- `EWS-low`: Marker genes from `tumor-cell-state-markers.tsv` that correspond to the `EWS-low` cell state. 
+- `EWS-high`: Marker genes from `tumor-cell-state-markers.tsv` that correspond to the `EWS-high` cell state. 
+- `proliferative` : Marker genes from `tumor-cell-state-markers.tsv` that correspond to the `proliferative` cell state. 
+Note that a lot of the Ewing sarcoma literature argues that `EWS-FLI1` high cells are proliferative. 
+- `aynaud-ews-high`: Marker genes from `gene_signatures/aynaud-ews-targets.tsv` that correspond to genes activated by `EWS-FLI1`. 
+- `wrenn-ews-low`: Marker genes from `gene_signatures/wrenn-nt5e-genes.tsv` that correspond to genes repressed by `EWS-FLI1`. 
+
+```{r, fig.height=7, fig.width=10}
+# pull out columns that have sum of marker gene expression 
+marker_gene_cols <- colnames(cluster_classification_df)[stringr::str_detect(colnames(cluster_classification_df), "_sum$")]
+
+# look at the expression of all gene sets across both libraries 
+marker_gene_cols |>
+  purrr::map(\(geneset){
+    cluster_classification_df |>
+      ggplot(aes(x = UMAP1, y = UMAP2, color = !!sym(geneset))) +
+      geom_point(alpha = 0.5, size = 0.1) +
+      facet_wrap(vars(library_id)) +
+      scale_color_viridis_c() +
+      theme(
+        aspect.ratio = 1
+      )
+  }) |> 
+  patchwork::wrap_plots(ncol = 2)
```

Comment:
I'm not sure I'd wrap these vs just print them one on each line, since they are a bit small in the HTML compared to other plots, and these are important ones to be able to see